src/items.zig

```diff
@@ -759,9 +769,48 @@ pub const Tool = struct { // MARK: Tool
 		self.durability -|= 1;
 		return self.durability == 0;
 	}
+	pub fn writeBytes(self: Tool, writer: *BinaryWriter) void {
+		var craftingGridMask: u32 = 0;
```

Comment:
Should use the right type `std.meta.Int(craftingGridSize)` isntead of arbitrarily using a `u32`